src/google/protobuf/compiler/objectivec/objectivec_helpers_unittest.cc

```diff
@@ -242,6 +243,135 @@ TEST(ObjCHelperDeathTest, TextFormatDecodeData_Failures) {
 }
 #endif  // PROTOBUF_HAS_DEATH_TEST
 
+class TestLineCollector : public LineConsumer {
+ public:
+  TestLineCollector(std::vector<std::string>* inout_lines,
+                    const std::string* reject_line = nullptr,
+                    bool skip_msg = false)
+    : lines_(inout_lines), reject_(reject_line), skip_msg_(skip_msg) {}
+
+  bool ConsumeLine(const StringPiece& line, std::string* out_error) override {
+    if (reject_) {
+      if (*reject_ == line) {
+        if (!skip_msg_) {
+          *out_error = std::string("Rejected '") + *reject_ + "'";
+        }
+        return false;
+      }
+    }
+    if (lines_) {
+      lines_->emplace_back(line);
+    }
+    return true;
+  }
+
+ private:
+  std::vector<std::string>* lines_;
+  const std::string* reject_;
+  bool skip_msg_;
+};
+
+const int kBlockSizes[] = {-1, 1, 2, 5, 64};
```

Comment:
Any particular reasons for these blocksizes? I know that -1 is default, but others?

Reply:
just some chunks to ensure the line buffering works out correctly.